src/server/plugins/router.ts

```diff
@@ -60,6 +60,34 @@ export default {
         }
       })
 
+      server.route<{ Payload: { 'cookies[additional]': string } }>({
+        method: 'post',
+        path: '/help/cookie-preferences',
+        handler(request, h) {
+          const decision = request.payload['cookies[additional]']
+
+          if (decision === 'yes') {
+            request.yar.set('cookieConsent', true)
+          } else if (decision === 'no') {
+            request.yar.set('cookieConsent', false)
+          } else {
+            throw Boom.badRequest('Unknown cookie preference')
+          }
+
+          request.yar.flash('cookieConsentUpdated', true, true)
```

Comment:
Did you see the [`cookies[hide]` example](https://design-system.service.gov.uk/components/cookie-banner/server-side-confirmation/) after updating consent?

Rather than use Yar we can just save each preference name/value in the same cookie

It also means when JavasScript is available we can write to these simple values client-side, preventing us having to POST the preferences form and navigate away from the page

E.g. Given this default cookie preferences object:

```js
const cookies = {
  essential: true,
  analytics: false,
  hide: false
}
```

## Accept analytics cookies
Clicking the accept button:

```html
<button name="cookie[analytics]" value="true">
  Accept analytics cookies
</button>
```

Updates the preferences:

```js
const cookies = {
  essential: true,
  analytics: true,
  hide: false
}
```

## Hide cookie message
Clicking the hide button:

```html
<button name="cookie[hide]" value="true">
  Hide cookie message
</button>
```

Updates the preferences:

```js
const cookies = {
  essential: true,
  analytics: true,
  hide: true
}
```

Reply:
Nice, I was speaking with Dave about something similar earlier.

`essential: true` is there a reason that one is tracked separately, given the user can't influence it (it will never be false).

Reply:
Yeah, sorry you missed this earlier

>essential: true is there a reason that one is tracked separately, given the user can't influence it (it will never be false)?

I've added `essential` only for completeness, it's not needed

Reply:
Sounds good. Moving this logic into a separate PR that will come later.